conference/orders.py

```diff
@@ -45,3 +50,138 @@ def next_order_code_for_year(year):
     # if there are no current codes, return the first one
     template = ORDER_CODE_TEMPLATE
     return template % {'year_two_digits': year % 1000, 'sequential_id': '0001'}
+
+
+def create_order(for_user, for_date, fares_info, calculation, coupon=None):
+    """
+    We assume that the data passed to this function is already sanitised,
+    ie. we assume that the calculation of discount is correct, and we just
+    check if the discount_code is valid, w/o changing how it should apply.
+    """
+    assert isinstance(calculation, OrderCalculation)
+
+    if coupon:
+        assert isinstance(coupon, Coupon)
+
+    fares = get_available_fares_as_dict(for_date)  # caching
+    with transaction.atomic():
+        # NOTE(artcz): Using Order().save() instead of Order.objects.create
+        # because .create provides/used to privde a different API.
+        # If we get rid of that dependency in other parts of the system we
+        # should be able to use .create just fine.
+        order = Order(
+            uuid=str(uuid.uuid4()),
+            user=for_user.assopy_user,
+            code=next_order_code_for_year(timezone.now().year)
+            # create a shell of an order without details
+        )
+        order.save()
+
+        for fare_code, amount in fares_info.items():
+            fare = fares[fare_code]
+            vat = fare.vat_set.all()[0]
```

Comment:
Not really related to this PR, but could a fare have multiple VAT rates?

Reply:
I think that's mostly because (for some reason) relationship is `Vat.fares::m2m -- VatFare -> Fare`
(where VatFare is `through` on m2m field)